pkg/k8s/apis/cilium.io/v2alpha1/lbipam_types.go

```diff
@@ -82,11 +92,22 @@ type CiliumLoadBalancerIPPoolSpec struct {
 	Disabled bool `json:"disabled"`
 }
 
-// CiliumLoadBalancerIPPoolCIDRBlock describes a single CIDR block.
-type CiliumLoadBalancerIPPoolCIDRBlock struct {
+type AllowFirstLastIPType string
+
+const (
+	AllowFirstLastIPNo  AllowFirstLastIPType = "no"
+	AllowFirstLastIPYes AllowFirstLastIPType = "yes"
+)
+
+// CiliumLoadBalancerIPPoolIPBlock describes a single IP block.
+type CiliumLoadBalancerIPPoolIPBlock struct {
 	// +kubebuilder:validation:Format=cidr
-	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Optional
 	Cidr IPv4orIPv6CIDR `json:"cidr"`
+	// +kubebuilder:validation:Optional
+	Start string `json:"start"`
+	// +kubebuilder:validation:Optional
+	Stop string `json:"stop"`
```

Comment:
Added the `omitempty`. I am holding off on the validation pattern. While I do understand, the reasoning, such regexes are near impossible to review/verify. They are also not very user friendly since kubectl will then spit back out: `{field} does not match {regex}`, which is huge and equally unreadable there. We have validation in place on the consuming side and report back bad IPs via the status (also not ideal). I would really like to see an improved version of this with https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook at some point.